### PR TITLE
Scripting system fixes/additions

### LINKED ISSUE #612: [Script] Turning the quest in doesn't spawn the mob.
**Describe the bug**
Handing in [Digging Through the Dirt](https://classic.wowhead.com/quest=254/digging-through-the-dirt) at [Eliza's Grave Dirt](https://classic.wowhead.com/object=51708/elizas-grave-dirt) doesn't spawn [Eliza](https://classic.wowhead.com/npc=314/eliza). 

**Steps to reproduce**
1. Add quest Bride of the Embalmer [ID 253] (Must have it in order to interact with Eliza's Dirt)
2. Turn in Digging through the Dirt

**Expected behavior**
Eliza [ID 314) should spawn a few seconds after Digging Through the Dirt is completed.

**Screenshots and proof**
While I couldn't find any screenshot of Eliza in the alpha catalogue, I base my report on the following:
1. Low IDs of all things associated with the quest (Quest ID 254), NPC (ID 314) Quest item (ID 2382) and game object (51708)
2. Lower ID of Eliza than that of [Stitches](https://classic.wowhead.com/npc=412/stitches), who appears earlier in the quest chain (most likely has a higher ID because of the event associated with this mob)
3. There are screenshots of Stitches' event (which is more complicated than spawning Eliza) working in patch 0.5.5. If this event was fixed at that time, the likelyhood of Eliza spawn being done is very high.

### PR DESCRIPTION
**Pull request by mindphluxnet**
Closes #612  (and also fixes Eliza's display_id)

Includes:

- moves ScriptHandler from ObjectManager to CreatureManager and GameObjectManager as it's not needed on everything
- improves handle_script_command_temp_summon_creature - it now spawns a temporary mob with no loot for despawn_type == 0. This is still not perfect, of course, but CreatureBuilder.create() simply doesn't know a despawn mode unless I am blind.
- fixes unique summon count which wasn't really working before
- **NEW** implements handle_quest_complete spell effect used by some quest items. Quest handling needs to be changed for this to actually do anything except sending the player the "objective complete" message.
- **NEW** implements script_command_start_movement using the new (waypoint) movement system

### CODE REVIEW

**Comment by GrenderG** [comment]
With missing despawn modes, you mean this (from VMaNGOS)?
```cpp
enum TempSummonType
{
    TEMPSUMMON_TIMED_OR_DEAD_DESPAWN          = 1,             // despawns after a specified time (out of combat) OR when the creature disappears
    TEMPSUMMON_TIMED_OR_CORPSE_DESPAWN        = 2,             // despawns after a specified time (out of combat) OR when the creature dies
    TEMPSUMMON_TIMED_DESPAWN                  = 3,             // despawns after a specified time
    TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT    = 4,             // despawns after a specified time after the creature is out of combat
    TEMPSUMMON_CORPSE_DESPAWN                 = 5,             // despawns instantly after death
    TEMPSUMMON_CORPSE_TIMED_DESPAWN           = 6,             // despawns after a specified time after death
    TEMPSUMMON_DEAD_DESPAWN                   = 7,             // despawns when the creature disappears
    TEMPSUMMON_MANUAL_DESPAWN                 = 8,             // despawns when UnSummon() is called
    TEMPSUMMON_TIMED_COMBAT_OR_DEAD_DESPAWN   = 9,             // despawns after a specified time (in or out of combat) OR when the creature disappears
    TEMPSUMMON_TIMED_COMBAT_OR_CORPSE_DESPAWN = 10,            // despawns after a specified time (in or out of combat) OR when the creature dies
};
```

**Comment by mindphluxnet** [comment]
> With missing despawn modes, you mean this (from VMaNGOS)?

yes

File: game/world/managers/objects/script/ScriptHandler.py [hunk 1]
```diff
@@ -353,21 +353,27 @@ def handle_script_command_respawn_gameobject(self, script):
...
-
-        if summoned_count <= script.datalong3:
+        can_summon = True
+
+        if script.datalong3 > 0:
+            summoned_count = 0
+            surrounding = MapManager.get_surrounding_units_by_location(script.source.location, script.source.map_id, script.source.instance_id, script.datalong4, False)
+            if len(surrounding) > 0:
+                for unit in surrounding:
+                    _unit = MapManager.get_surrounding_unit_by_guid(script.source, unit)
```
[the remark this answers is not in the document]
Comment (mindphluxnet) [reply game/world/managers/objects/script/ScriptHandler.py]: Done

File: game/world/managers/objects/script/ScriptHandler.py [hunk 2]
```diff
@@ -357,11 +357,10 @@ def handle_script_command_temp_summon_creature(self, script):
 
         if script.datalong3 > 0:
             summoned_count = 0
-            surrounding = MapManager.get_surrounding_units_by_location(script.source.location, script.source.map_id, script.source.instance_id, script.datalong4, False)
-            if len(surrounding) > 0:
-                for unit in surrounding:
-                    _unit = MapManager.get_surrounding_unit_by_guid(script.source, unit)
-                    if _unit and _unit.creature_template.entry == script.datalong:
+            surrounding_units = MapManager.get_surrounding_units_by_location(script.source.location, script.source.map_id, script.source.instance_id, script.datalong4, False)
+            if len(surrounding_units) > 0:
```
Comment (GrenderG) [inline game/world/managers/objects/script/ScriptHandler.py:361]: I think this check can be removed, might be redundant.

**Pull request merged**